neuronpp/core/cells/spine_cell.py

```diff
@@ -78,3 +120,369 @@ def _connect_necks_rand_uniform(neck: Sec, sections):
+        """
+        Add spines to a section of a dedrite. There are two possibilities:
+        1) spines are added uniformly every n_spines/section_length,
+        2) spines positions on the dendrite's section are drawn
+        from the uniform distribution.
+        
+        :param section:
+           section
+        :param spine_name:
+           string attached to name of every head and neck
+        :param n_spines:
+           number of spines
+        :param head_diam:
+           diameter of the spine head
+        :param head_len:
+           length of the spine_head
+        :param neck_diam:
+           diameter of the neck
+        :param neck_len:
+           length of the neck
+        :param add_pas:
+           add passive mechanism
+        :param u_random:
+           if int draw spine position from the uniform distribution
+        """
+        if not isinstance(section, Sec):
+            section = Sec(section)
+        if isinstance(u_random, int):
+            target_locations = np.random.uniform(0., 1., n_spines).tolist()
+        else:
+            target_locations = np.linspace(0., .99, n_spines).tolist()
+
+        self._add_spines_to_section(section, spine_name,
+                                    target_locations, head_diam,
+                                    head_len, neck_diam, neck_len,
+                                    E_pas, g_pas, ra, cm,
+                                    add_pas=add_pas)
+        return target_locations
+
+    def _add_spines_to_section(self, section, spine_name,
+                               target_location, head_diam,
+                               head_len, neck_diam, neck_len,
+                               E_pas, g_pas, ra, cm, add_pas=True):
+        name = section.name
+        if not isinstance(target_location, list):
+            target_location = [target_location]
+        for i, location in enumerate(target_location):
+            head = self.add_sec(name="%s_%s_head[%d]" % (name, spine_name, i),
+                                diam=head_diam, l=head_len, nseg=2,
+                                E_rest=E_pas, ra=ra, cm=cm,
+                                g_pas=g_pas, add_pas=add_pas)
+            neck = self.add_sec(name="%s_%s_neck[%d]" % (name, spine_name, i),
+                                diam=neck_diam, l=neck_len, nseg=1,
+                                E_rest=E_pas, ra=ra, cm=cm,
+                                g_pas=g_pas, add_pas=add_pas)
+            self.heads.append(head)
+            self.necks.append(neck)
+            spine = Spine(head, neck, self, "%s_spine_%s[%d]" % (name,
+                                                                 spine_name,
+                                                                 i))
+            self.spines.append(spine)
+            self.connect_secs(source=neck, target=section, source_loc=location,
+                              target_loc=0.0)
+
+    def find_sections_with_mech(self, mech_name):
```

Comment:
if it's a public method - please add at least short docstring as well

Reply:
The method is not finding sections by `mech_name` but rather spines by `mech_name` of their parents.

I think it should be renamed - to be self explanatory eg. `get_spines_by_section` or something similar.

Reply:
Did you want to get spines with `mech_name` or spines which parent has `mech_name` ? Because here you do the latter.